src/Compilers/Core/Portable/DiagnosticAnalyzer/SourceTextValueProvider.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Provides custom values associated with <see cref="SourceText"/> instances using the given 'computeValue' delegate.
```

Comment:
I suspect the quotes in 'computeValue' are not needed, but I'm not completely aware of our conventions.